spelling corrections etc

PublicHealthDataGeek opened the pull request:
some spelling corrections, some comments for RL to consider, some definitions added for clarity, some comments added to code.

PublicHealthDataGeek committed:
spelling corrections etc

some spelling corrects, some comments for RL to consider, some definitions added for clarity, some comments added to code

Robinlovelace commented on 02-basics.Rmd:
````diff
@@ -7,9 +7,9 @@ This chapter may seem tedious for people who just want to crack on and load-in d
+You should see an 'R console pane' like that displayed in Figure \@ref(fig:console).
 
-```{r console, echo=FALSE, fig.cap='The R console in RStudio. You should see something like this when you first open RStudio. Tapping Enter at this point will trigger the autocompletion of the command `install.packages("stats19").'}
+```{r console, echo=FALSE, fig.cap='The R console pane in RStudio. You should see something like this when you first open RStudio. Having entered the text 'install.packages("stats', tapping 'Enter' at this point will trigger the autocompletion of the command `install.packages("stats19").'}
````
This is definitely much better. 'Tapping' is a funny word, would 'typing Enter' be better?... Probably not.

PublicHealthDataGeek replied:
Yes it would

Robinlovelace commented on 02-basics.Rmd:
```diff
@@ -20,7 +20,7 @@ You are ready to start running R code, by entering commands into the console.
 R can be understood as a giant calculator.
 If you feed the console arithmetic tasks, it will solve them precisely and instantly.
-Try typing the following for examples (note that `pi` is an inbuilt object) into the R console in RStudio and pressing `Enter` to make R run the code:
+Try typing the following examples (note that `pi` is an inbuilt object) into the R console in RStudio and pressing `Enter` to make R run the code.  The output of the code, when shown in this book, is preceeded by '##'.
```
That is a very good point you make about `##`.

PublicHealthDataGeek replied:
not sure if I need to comment or not to this...Just my lack of knowledge on how to use github again!

PublicHealthDataGeek replied:
:)

Robinlovelace commented on 02-basics.Rmd:
```diff
@@ -36,10 +36,10 @@ sqrt(361)
 However, a key benefit of R is that it is *object oriented*, meaning it stores values and complex objects such as data frames representing road casualties and processes them *in memory* (meaning that R is both fast and memory hungry when working with large datasets).
 
-The two most common ways of creating objects are with the the `<-` 'arrow' or `=` 'equals' assignment operators.^[
+The two most common ways of creating objects are using `<-` 'arrow' or `=` 'equals'.  These are known as assignment operators as they assign contents(?), such as numbers, to objects.^[
```
Thanks for this, definite improvement in terms of readability. 'they assign contents' is fine.

PublicHealthDataGeek replied:
:+1:

Robinlovelace commented on 02-basics.Rmd:
````diff
@@ -48,13 +48,13 @@ z = x + y
 ```
 
-The previous code chunk created three objects called `x`, `y` and `z` and showed how these objects can themselves be used to create additional objects.
+The previous code chunk created and stored three objects called `x`, `y` and `z` and showed how these objects can themselves be used to create additional objects.
````
:+1:

PublicHealthDataGeek replied:
:)

Robinlovelace commented on 02-basics.Rmd:
```diff
@@ -65,9 +65,9 @@ rm(x, y, z)
 
 What just happened?
-We have removed the objects using the R function `rm()`, which stands for 'remove'.
+We have removed the objects using the R function `rm()`, which stands for 'remove'. A function is an instruction or set of instructions for R to do something with what we give to that function.  What we give to the function are known as arguments.  Each function has set of arguments we can potentially give to it.  
```
Important clarification.

PublicHealthDataGeek replied:
:)

Robinlovelace commented on 02-basics.Rmd:
```diff
@@ -65,9 +65,9 @@ rm(x, y, z)
+We have removed the objects using the R function `rm()`, which stands for 'remove'. A function is an instruction or set of instructions for R to do something with what we give to that function.  What we give to the function are known as arguments.  Each function has set of arguments we can potentially give to it.  
 Technically speaking, we *passed the objects to arguments in the `rm()` function call*.
-In plain English, things that go inside the curved brackets that follow a function name are arguments.
+In plain English, things that go inside the curved brackets that follow a function name are the arguments.
```
How about:

```
In plain English, things that go inside the curved brackets following a function name are the function's *arguments*.
```

Even more specific.

PublicHealthDataGeek replied:
Yes - even better!

Robinlovelace commented on 02-basics.Rmd:
```diff
@@ -132,13 +132,13 @@ ncol(crashes)
 We will use larger datasets, with thousands of rows and tens of columns, in later chapters but for now it's good to 'start small' to understand the basics of R.
 
-## Subsetting by index or name
+## Subsetting by index or name CHANGE TITLE OR BE CLEARER WHEN YOU ARE SUBSETTING BY INDEX OR NAME IN THE TEXT BELOW
```
Good point. How about just 'Subsetting'? 

Would

'Subsetting vectors' 

be any better?

PublicHealthDataGeek replied:
Would need to define vector which I mentioned below. If 2.3 was subsetting then would you merge it with 2.4 (which is also subsetting)?  I think you could keep as subsetting by index or name if you make it clearer in the text what 'index' and 'name' mean.

PublicHealthDataGeek replied:
ah looks like you have sorted it below!!!

Robinlovelace commented on 02-basics.Rmd:
```diff
@@ -132,13 +132,13 @@ ncol(crashes)
 
 Subsetting returns part of an R object. 
-It can be done by providing numbers representing the positions of the elements we want (e.g. the 2^nd^ element) or with a logical vector, with values associated with `TRUE` returned. 
+It can be done by providing numbers representing the positions of the elements we want (e.g. the 2^nd^ element) or with a logical vector *NEED TO EXPLAIN VECTOR, with values associated with `TRUE` or `FALSE` returned. 
```
Good point. How about:

```
The most basic type of R object is a *vector*, a sequence of values of the same type.
`x`, `y` and `z` were all *numeric vectors* with a length of 1; `casualty_type` is a *character vector* (because it contains letters that cannot be added) or length 3; `casualty_age` is a *numeric vector* of length three.
In R, subsets of vectors can be returned by providing numbers representing the positions of the elements we want (e.g. '2' representing selection of the 2^nd^ element) or with logical (`TRUE` or `FALSE`) values associated with the elements that should be selected. 
```

PublicHealthDataGeek replied:
Few more edits below- hope they make sense 


The most basic type of R object is a *vector*, a sequence of values of the same type, FOR EXAMPLE, TYPE COULD BE NUMERIC.
`x`, `y` and `z` were all *numeric vectors* with a length of 1; `casualty_type` is a *character vector* (because it contains letters that cannot be added) OF length 3; `casualty_age` is a *numeric vector* of length 3. CHANGED THREE TO 3 FOR COSNISTENCY
In R, subsets of vectors can be returned by providing numbers representing the positions of the elements we want WITHIN THE VECTOR (e.g. '2' representing selection of the 2^nd^ element) or with logical (`TRUE` or `FALSE`) values associated with the elements that should be selected.

PublicHealthDataGeek replied:
SUGGESTED WORDING CHANGES BELOW

The most basic type of R object is a *vector*, a sequence of values of the same type FOR EXAMPLE, A SET OF NUMBER.
`x`, `y` and `z` were all *numeric vectors* with a length of 1; `casualty_type` is a *character vector* (because it contains letters that cannot be added) OF length 3; `casualty_age` is a *numeric vector* of length 3(CHANGED TO THREE FOR CONSISTENCY).


In R, subsets of vectors can be returned by providing numbers representing the positions of the elements IN THE VECTOR  we want (e.g. '2' representing selection of the 2^nd^ element) or with logical (`TRUE` or `FALSE`) values associated with the elements that should be selected.

Robinlovelace commented on 02-basics.Rmd:
```diff
@@ -214,7 +214,7 @@ crashes$casualty_age[crashes$vehicle_type == "tank"] = 61
 ## Dealing with NAs and recoding
 
-R objects can have a value of NA. This is how R represents missing data.
+R objects can have a value of NA. NA is how R represents missing data.
```
:+1:

Robinlovelace commented on 02-basics.Rmd:
```diff
@@ -260,7 +260,7 @@ as.matrix(crashes)
 Often it is useful to 'recode' values.
 In the raw STATS19 files, for example, -1 means NA.
-There are many ways to recode values in R, the simplest and most mature of which is the use of factors, as shown below:
+There are many ways to recode values in R, the simplest and most mature of which is the use of factors DEFINE WHAT A FACTOR IS, as shown below:
```
How about:

```
 There are many ways to recode values in R, the simplest and most mature of which is the use of factors, which are whole numbers representing characters, as shown below: 
```

On second thoughts, is this even a useful example? It is a bit confusing I think so it may be worth scrapping it IMO...

PublicHealthDataGeek replied:
There are quite a few use cases for using factors in STATS19 eg recoding vehicles to simpler categories so I think it is helpful to have a look at factors.  I think the wording you have suggested is better - perhaps worth saying something that factors are usually used for categorical data - i think that would help - would a male/femae/other example would be more helpful?

PublicHealthDataGeek replied:
I think the wording is now good.  A better example might be to use gender e.g male, female, other as that is more relevent and highlights how we tend to use factors - to code categorical data.  Perhaps add a sentence saying that are usually used for categorical variables.

Robinlovelace commented on 02-basics.Rmd:
```diff
@@ -275,7 +275,7 @@ z_charcter
 ## Saving R objects
 
-You can also save individual R objects in the RDS format.
+You can also save individual R objects in the RDS format. The RDS format is the ? data format for R. (? is this the right term?)
```
Yes that's the right term. How about the following?

``` 
You can also save individual R objects in the Rds format. The Rds format is R's native data format, meaning that any R object can be saved as an Rds file.
```

PublicHealthDataGeek replied:
Perhaps you could reference it to Excel syaing that 'in the same way that .xls is the native Excel data format.  ?

Robinlovelace replied:
good idea

PublicHealthDataGeek replied:
Suggest add the wording below or similar:  
This is comparable to Excel where the native data format is xls.

Robinlovelace commented on 02-basics.Rmd:
````diff
@@ -291,7 +291,7 @@ identical(crashes, crashes2)
 
 ```{r readr-write, eval=FALSE}
-readr::write_csv(crashes, "crashes.csv")
+readr::write_csv(crashes, "crashes.csv") # This code tells R to use the write_csv function from the readr package
````
:+1:

Robinlovelace reviewed (changes requested):
Great set of suggestions. I've suggested a few tweaks before merging. Thanks Caroline!

Robinlovelace commented:
Thanks for all the replies Caroline, it seems we're on the same wavelength. Do you want to have one more go at changing a few things? If so let me know when you're done (you can edit this version, let me know if you have any issues finding out how) and I'll merge this, can always tidy-up after :broom:

PublicHealthDataGeek commented:
Hi Robin, thanks for the feedback.  I dont know what quite happened - I put in various replies then looked at help and thought that I had lost all my comments but seeing your reply 5 mins ago it looks like you might have got them...?  anyway, I had gone through again and done them again and so you might get them again.  I am not sure how to 'changes things and edit this version'?  Do I just refresh the book and then go through and edit again?

Robinlovelace commented:
Hi @PublicHealthDataGeek I can see all of your comments but some of them show as 'resolved' if you edit sections as shown below, nothing has been lost.

![image](https://user-images.githubusercontent.com/1825120/95833693-e87e0c00-0d33-11eb-91d3-b087fc10e2af.png)

Robinlovelace commented:
Can you try making a few more edits here @PublicHealthDataGeek?: https://github.com/PublicHealthDataGeek/rrsrr/edit/patch-3/02-basics.Rmd?pr=%2Fsaferactive%2Frrsrr%2Fpull%2F6

Partly for interest and partly to make the editing process quicker when I merge this PR.

PublicHealthDataGeek committed:
updated comments as using the link

Hi Robin - this is me updating the outstanding wording as per seplling corrections etc #6 using the link you posted... 
https://github.com/PublicHealthDataGeek/rrsrr/edit/patch-3/02-basics.Rmd?pr=%2Fsaferactive%2Frrsrr%2Fpull%2F6

PublicHealthDataGeek commented:
HI Robin - used the link - it has posted it under your comment yestereday.  Then I clicked on 'request review'above so fingers crossed it all links seemlessly and you can see the futher changes I made to incorporate our discussion.....

Robinlovelace commented on 02-basics.Rmd:
```diff
@@ -275,7 +278,7 @@ z_charcter
 ## Saving R objects
 
-You can also save individual R objects in the RDS format. The RDS format is the ? data format for R. (? is this the right term?)
+You can also save individual R objects in the RDS format. The RDS format is the data format for R, meaning that any R object can be saved as an RDS file. This is equivalent to xls being the data format for Excel. 
```
Many thanks for the update @PublicHealthDataGeek !

Robinlovelace commented:
Great news @PublicHealthDataGeek it works, this is what I see: 

![image](https://user-images.githubusercontent.com/1825120/95971969-cc4c9e80-0e09-11eb-9bda-f1ebc6c34591.png)

Robinlovelace reviewed (approved):
Many thanks for the updates Caroline, way better.

Robinlovelace commented:
Good practice:

![image](https://user-images.githubusercontent.com/1825120/95972151-0453e180-0e0a-11eb-860b-909035286614.png)

Robinlovelace commented:
Heads-up @PublicHealthDataGeek I've just cleaned up Chapter 2. Please take a look at the new shiny results here: https://saferactive.github.io/rrsrr/basics.html
